File: assignments/binary_art_converter/javascript/binary_art_converter_test_runner.js
const assert = require('assert');
const { BinaryArtConverter } = require('./binary_art_converter.js'); // Assuming student file

function runTests() {
    const solver = new BinaryArtConverter();
    const testCases = [
        { input: "Hi", expected: "⚫⚪⚫⚫⚪⚫⚫⚫\n⚫⚪⚪⚫⚫⚪⚫⚪" },
        { input: "A", expected: "⚫⚪⚫⚫⚫⚫⚫⚪" },
        { input: "z", expected: "⚫⚪⚪⚪⚪⚫⚪⚫" },
        { input: "", expected: "" },
        { input: "1", expected: "⚫⚫⚪⚪⚫⚫⚫⚪" },
        { input: "Hello", expected: "⚫⚪⚫⚫⚪⚫⚫⚫\n⚫⚪⚪⚫⚫⚪⚫⚪\n⚫⚪⚪⚫⚪⚪⚫⚫\n⚫⚪⚪⚫⚪⚪⚫⚫\n⚫⚪⚪⚫⚪⚪⚪⚪" }
    ];

    let passedCount = 0;
    let report = [];
    console.log("===== Starting Automated Tests for BinaryArtConverter (JavaScript) =====");

    testCases.forEach((tc, i) => {
        let actual = "";
        const caseIdentifier = `Test Case #${i + 1}: Input "${tc.input}"`;
        try {
            actual = solver.convertToBinaryArt(tc.input);
            // Use strictEqual for exact string comparison including newlines
            assert.strictEqual(actual, tc.expected);
            report.push(`${caseIdentifier} -> ✅ PASS`);
            passedCount++;
        } catch (error) {
             // Show expected vs actual clearly on failure
             report.push(`${caseIdentifier} -> ❌ FAIL/ERROR:\n -> Expected:\n${tc.expected}\n ->      Got:\n${actual}`);
        }
    });

    console.log(report.join('\n\n'));
    console.log("\n===== Test Summary =====");
    console.log(`Total Tests: ${testCases.length}`); console.log(`Passed: ${passedCount}`); console.log(`Failed: ${testCases.length - passedCount}`); console.log("========================");
    console.log(`[SUMMARY]:${passedCount}:${testCases.length}`);
}

runTests();

File: assignments/binary_art_converter/python/binary_art_converter_test_runner.py
import sys
from binary_art_converter import BinaryArtConverter # Import the student's class

def run_tests():
    solver = BinaryArtConverter()
    test_cases = [
        ("Hi", "⚫⚪⚫⚫⚪⚫⚫⚫\n⚫⚪⚪⚫⚫⚪⚫⚪"),
        ("A", "⚫⚪⚫⚫⚫⚫⚫⚪"),
        ("z", "⚫⚪⚪⚪⚪⚫⚪⚫"),
        ("", ""),
        ("1", "⚫⚫⚪⚪⚫⚫⚫⚪"),
        ("Hello", "⚫⚪⚫⚫⚪⚫⚫⚫\n⚫⚪⚪⚫⚫⚪⚫⚪\n⚫⚪⚪⚫⚪⚪⚫⚫\n⚫⚪⚪⚫⚪⚪⚫⚫\n⚫⚪⚪⚫⚪⚪⚪⚪")
    ]
    passed_count = 0
    report = []
    print("===== Starting Automated Tests for BinaryArtConverter (Python) =====")

    for i, (input_text, expected) in enumerate(test_cases):
        actual = ""
        case_identifier = f'Test Case #{i + 1}: Input "{input_text}"'
        try:
            actual = solver.convert_to_binary_art(input_text)
            report.append(case_identifier)

            if actual == expected:
                report.append(f" -> Expected:\n{expected}")
                report.append(f" ->      Got:\n{actual}")
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(f" -> Expected:\n{expected}")
                report.append(f" ->      Got:\n{actual}")
                report.append(" -> ❌ FAIL\n")
        except Exception as e:
            report.append(case_identifier)
            report.append(f" -> ❌ ERROR: An exception occurred - {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary ====="); print(f"Total Tests: {len(test_cases)}"); print(f"Passed: {passed_count}"); print(f"Failed: {len(test_cases) - passed_count}"); print("========================")
    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    sys.path.append('.')
    run_tests()